### PR TITLE
UI tests for secure local storage sample app

### PR DESCRIPTION
**Pull request by as666499**
adding UI tests for sample app.

- creating and updating image and string objects in app segment and app user segment
- deletion of objects in both segments

### CODE REVIEW

**Comment by as666499** [comment]
> 1). Remove all Local string to Global file 2). Create property for “[XCUIApplication alloc] init]” to reduce minimum number of initialization

incorporated feedback

File: StorageApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -239,20 +304,22 @@
...
 			outputPaths = (
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASFoundation.framework",
+				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASProximity.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASStorage.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MASUI.framework",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-StorageApp/Pods-StorageApp-frameworks.sh\"\n";
+			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-StorageApp/Pods-StorageApp-frameworks.sh\"\n";
```
Comment (ysd24) [inline StorageApp.xcodeproj/project.pbxproj:322]: Do not checkin the xcworkspace changes created by cocoapods. Please revert back to old state. Add your UITests files and checkin only the relevant xcodeproj/pbxproj changes.

File: StorageApp.xcodeproj/xcshareddata/xcschemes/StorageApp.xcscheme [hunk 2]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline StorageApp.xcodeproj/xcshareddata/xcschemes/StorageApp.xcscheme:1]: Do not checkin this file. These changes are user specific. Please revert these changes.

File: StorageApp.xcodeproj/xcshareddata/xcschemes/StorageAppUITests.xcscheme [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline StorageApp.xcodeproj/xcshareddata/xcschemes/StorageAppUITests.xcscheme:1]: Do not checkin this file. These changes are user specific. Please revert these changes.

File: StorageApp.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline StorageApp.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: Do not checkin this file. These changes are user specific. Please revert these changes.

File: StorageApp/StorageTableViewController.m [hunk 5]
```diff
@@ -13,8 +13,8 @@
 #import <MASFoundation/MASFoundation.h>
 #import <MASStorage/MASStorage.h>
 
-static NSString *SampleUser = @"YOU USERNAME";
-static NSString *SampleUserPassword = @"YOU PASSWORD";
+static NSString *SampleUser = @"admin";
```
Comment (ysd24) [inline StorageApp/StorageTableViewController.m:16]: Do not checkin these sensitive data changes.

File: StorageAppUITests/StorageAppUITests.m [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+//  StorageAppUITests.m
+//  StorageAppUITests
+//
+//  Created by Ashwin Kumar on 10/12/21.
+//  Copyright © 2021 CA Technologies. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+
+#define INTERRUPT_MONITOR_DESCRIPTION @"TestInitialScreenHandler"
```
Comment (ysd24) [inline StorageAppUITests/StorageAppUITests.m:11]: Please take this input from an external json. So that all the tests can reuse this data. And it can be tracked in one place.

File: StorageAppUITests/StorageAppUITests.m [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (XCUIApplication*) initializeApp {
+    XCUIApplication *app = [[XCUIApplication alloc] init];
+    [self addUIInterruptionMonitorWithDescription:INTERRUPT_MONITOR_DESCRIPTION handler:^BOOL(XCUIElement * _Nonnull interruptingElement) {
+        
+        // handling permissions popup
```
Comment (ysd24) [inline StorageAppUITests/StorageAppUITests.m:53]: Please follow our commenting style to be uniform across, for example :
//
// Invoke protected endpoint
//

File: msso_config.json [hunk 8]
```diff
@@ -1 +1,136 @@
-Add msso_config from your server here
\ No newline at end of file
+{
```
Comment (ysd24) [inline msso_config.json:1]: Do not checkin the msso_config.json

File: StorageAppUITests/StorageAppUITestsLaunchTests.m [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    [app launch];
+
+    // Insert steps here to perform after app launch but before taking a screenshot,
+    // such as logging into a test account or navigating somewhere in the app
+
+    XCTAttachment *attachment = [XCTAttachment attachmentWithScreenshot:XCUIScreen.mainScreen.screenshot];
+    attachment.name = @"Launch Screen";
+    attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [self addAttachment:attachment];
+}
```
Comment (ysd24) [inline StorageAppUITests/StorageAppUITestsLaunchTests.m:36]: Please add some negative tests as well.

**Review by ysd24 — COMMENTED** [review]
Please use develop branch as base for any changes. Then move the changes to master.

**Comment by as666499** [comment]
Created a new PR with these changes https://github.com/CAAPIM/Sample-App-MAS-iOS-Secure-Local-Storage/pull/12
